### ISSUE TITLE
Round-off error in `RegularCartesianGrid` produces ranges of wrong length

### ISSUE THREAD

**Issue by ali-ramadhan** [climate-machine/Oceananigans.jl#480]
Noticed this because `test_diffusion_cosine` was failing in PR #462.

We should have that `length(zC) == Nz` but
```julia
julia> Nz = 16; Lz = π/2; Δz = Lz / Nz; zC = -Lz+Δz/2:Δz:-Δz/2; length(zC)
16

julia> Nz = 32; Lz = π/2; Δz = Lz / Nz; zC = -Lz+Δz/2:Δz:-Δz/2; length(zC)
32

julia> Nz = 64; Lz = π/2; Δz = Lz / Nz; zC = -Lz+Δz/2:Δz:-Δz/2; length(zC)
63

julia> Nz = 128; Lz = π/2; Δz = Lz / Nz; zC = -Lz+Δz/2:Δz:-Δz/2; length(zC)
127

julia> Nz = 256; Lz = π/2; Δz = Lz / Nz; zC = -Lz+Δz/2:Δz:-Δz/2; length(zC)
255
```

Looks like a scary effect of round-off error to me.

I'll make the switch to using the `range` function with a fixed `length` and add a test to check for this.

**Comment by glwagner** [comment]
I think ‘range’ is clearer anyways!

**Comment by glwagner** [comment]
Is this fixed?

**Comment by ali-ramadhan** [comment]
Yes it's been fixed in PR #462 (had to fix it as the issue arose once the k index was reversed), just need to merge it.
